### PR TITLE
Random quest ID

### PR DESCRIPTION
**Pull request by KatatsumuriPan**
Now newly created quest ID will be random.
This prevents merge conflicts when multiple people try to add or remove quests simultaneously.

Changing the type of quest ID from int to UUID is pending for compatibility.

I won't apply this change to quest-line ID and other IDs until https://github.com/CleanroomMC/BetterQuesting/pull/103 is merged.

### CODE REVIEW

File: src/main/java/betterquesting/api2/storage/RandomIndexDatabase.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package betterquesting.api2.storage;
+
+import java.util.Random;
+
+public class RandomIndexDatabase<T> extends AbstractDatabase<T> {
+
+    private final Random random = new Random();
+
+    @Override
+    public synchronized int nextID() {
```
Comment (WaitingIdly) [inline src/main/java/betterquesting/api2/storage/RandomIndexDatabase.java:10]: this is only called when creating a new quest, but not on other situations in which a quest may be created, such as `ToolboxToolCopy.java#getNextIDs` (although that appears to be a unique case due to its bulk creation).

**Review by WaitingIdly — CHANGES_REQUESTED** [review]
problems:
- similar logic should also apply to questlines
- since the questID is absurdly long now, should change `PanelButtonQuest.java:103` to *not* display the questID, regardless of if advancedInfo is enabled
- again, absurdly long so it should be less publicly visible, should change `EventHandler.java:142` so sharing the quest to chat doesnt include the questID.
- really long, want an easy way to copy - `GuiQuestLines.java:354` is the place where we get the popup for "Share to Chat", should add an option below that for "Copy ID" which copies the quest ID to the clipboard

question:
what is the logic that leads to continuing to use an int instead of switching to using a UUID, as was done in https://github.com/GTNewHorizons/BetterQuesting/pull/89?

**Comment by KatatsumuriPan** [comment]
> problems:
> * similar logic should also apply to questlines

I won't apply this change to quest-line ID and other IDs until https://github.com/CleanroomMC/BetterQuesting/pull/103 is merged.

> question:
> what is the logic that leads to continuing to use an int instead of switching to using a UUID, as was done in https://github.com/GTNewHorizons/BetterQuesting/pull/89?

I think the only advantage of UUID is their randomness.
Considering the cost and trouble, I don't think there is any benefit to changing to UUID.

**Comment by Rongmario** [comment]
Closing in favour of #113. Thank you for all the work!

**Pull request closed without merging**